src/components/Courses/Courses.jsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from 'react';
+
+import { CourseCard, SearchBar } from '../../components';
+import { Button, FlexContainer, BorderedFlexContainer } from '../../common';
+
+import { getAuthorsNamesById } from '../../helpers';
+import { ADD_NEW_COURSE_BUTTON_TEXT } from '../../constants';
+
+import styled from 'styled-components';
+
+const StyledCourses = styled(BorderedFlexContainer).attrs({ as: 'main' })`
+	border-bottom-left-radius: 0;
+	border-bottom-right-radius: 0;
+	flex-grow: 1;
+
+	.automargin button {
+		margin: 0 auto;
+	}
+`;
+
+const Courses = ({ handleAddNewCourseButtonClick, courses, authors }) => {
+	const [coursesToDisplay, setCoursesToDisplay] = useState([...courses]);
```

Comment:
You are creating a copy of the courses list here, but it is not necessary. You can move this state logic to the parent component hence removing data copying and creating a pure component.

Reply:
I understood why you did this. 
You are changing the array itself, but you can just filter it out at line 47. like this:
`courses.filter((it) => it.name === <search text>).map(<visualisation>)`
or if you like it more you can do it within `map` method by returning `null` instead of html

(you can decide what to use yourself)